netlify.toml

```diff
@@ -1,3 +1,7 @@
+[build]
+command = "next build"
+publish = ".next"
```

Comment:
Note that updating this will cause errors when folks run `netlify dev` locally as the local environment has to accomodate http://localhost:8888. See https://discord.com/channels/714698561081704529/928693344358514698/1137009113411358770

Reply:
@nickytonline I documented this issue in a comment in the edge function https://github.com/open-sauced/insights/pull/1400/files#diff-ac00563b2d3b101fc9c19ca1382e6b3de979cc00debffb668ccf08544941faa8R40-R42

Removing this config line doesn't seem to resolve the issue though. Is there a way to have the next.js app and the edge-functions running on the same port? I know that when this was a serverless function that lived in the next.js `/pages/api` directory it wouldn't have this issue because it would be routed by the next application and could thus run on the same port, but there didn't seem to be a way to use edge functions on netlify without taking them out of the next app and putting them in the `netlify` folder.

Do you have any advice on how to make this more ergonomic?

Reply:
I saw the comment. Hard-coding may work temporarily for testing this in however you are currently set up, but to run Netlify Edge Functions, you need to run `netlify dev`. Next.js will run on port `3000` and the Netlify CLI will proxy requests from port `3000` to port `8888`. So for local development, everything runs on http://localhost:8888.

This wasn't necessary before the introduction of Edge functions in this PR. That's why I was pointing to the discussion in Discord.

Reply:
Yah, i didn't realize `netlify dev` was meant to proxy that way. It's been broken up until now with our setup. It looks like the `middleware.ts` is interfering with a lot of requests when hitting the app through `localhost:8888` but not when hitting the app through `localhost:3000` I'm going to try and dig into why

Reply:
The middleware should be running on the edge and `netlify dev` handles that, so not sure what's up. 🤔

Reply:
Yeah, the middleware is running, but I'm noticing that when accessing the site through http://localhost:8888 all requests to assets in the `_next` folder are getting redirected to `/feed`

<img width="1088" alt="CleanShot 2023-08-11 at 11 02 52@2x" src="https://github.com/open-sauced/insights/assets/555044/776eabc3-4290-4e6f-944b-478dcfe74f24">

For some reason this doesn't happen when accessing the site through http://localhost:3000.

If i disable the middleware by renaming the `middleware.ts` file to `middleware.bak.ts` the problem goes away.

For some reason, when running through the netlify dev proxy that middleware is being extra greedy and matching requests to `/_next/` inspite of the fact that we have a [matcher configured](https://github.com/open-sauced/insights/blob/beta/middleware.ts#L33-L35)

@brandonroberts do you have any ideas about this?

Reply:
I'm gonna pull down your branch and see what's up. 👀

Reply:
This actually seems fine what you've done in the netlify.toml as it's still proxying and this gets rid of the issue I mentioned the other day @brandonroberts with the infinite redirect loop.

```
[dev]
command = "next dev -p 3001"
port = 3000
targetPort = 3001
```

So for running `netlify dev`, all good, and it you happen to run `npm run dev`, you're still port 3000.

I'm up and running on my local with this setup.

Reply:
Everything seems to be running fine on http://localhost:3000

![CleanShot 2023-08-11 at 19 52 52](https://github.com/open-sauced/insights/assets/833231/8c752e89-4413-4ac2-b72a-7a7065ea81a9)

Reply:
> This actually seems fine what you've done in the netlify.toml as it's still proxying and this gets rid of the issue I mentioned the other day @brandonroberts with the [infinite redirect loop](https://discord.com/channels/714698561081704529/928693344358514698/1137009113411358770).
> 
> ```
> [dev]
> command = "next dev -p 3001"
> port = 3000
> targetPort = 3001
> ```
> 
> So for running `netlify dev`, all good, and it you happen to run `npm run dev`, you're still port 3000.
> 
> I'm up and running on my local with this setup.

Yeah, i just pushed this fix up a couple hours ago. It resolves a lot of issues. I encountered a bug where the middleware wasn't always matching correctly. #1549 

What's strange is that in some cases (i.e. when I was running `netlify dev` before) it was being way too greedy with matching, and in other cases like in #1549 it's not matching at all.

Going to do some staging testing and then get this ready for review